pymeasure/instruments/lecroy/lecroyT3DSO1204.py

```diff
@@ -0,0 +1,829 @@
+    )
+
+    @property
+    def timebase(self):
+        """ Read timebase setup as a dict containing the following keys:
+            - "seconds_div": horizontal scale in seconds/div (float)
+            - "delay": interval in seconds between the trigger event and the reference position (float)
+            - "hor_magnify": horizontal scale in the zoomed window in seconds/div (float)
+            - "hor_position": horizontal position in the zoomed window in seconds (float)"""
+        tb_setup = {
+            "seconds_div": self.timebase_scale,
+            "delay": self.timebase_offset,
+            "hor_magnify": self.timebase_hor_magnify,
+            "hor_position": self.timebase_hor_position
+        }
+        return {k: float(v) for k, v in tb_setup.items()}
+
+    def timebase_setup(self, offset=None, scale=None, hor_magnify=None, hor_position=None):
+        """ Set up timebase. Unspecified parameters are not modified. Modifying a single parameter
+        might impact other parameters. Refer to oscilloscope documentation and make multiple
+        consecutive calls to timebase_setup if needed.
+
+        :param offset: horizontal scale per division in seconds/div.
+        :param scale: interval in seconds between the trigger event and the reference position.
+        :param hor_magnify: horizontal scale in the zoomed window in seconds/div.
+        :param hor_position: horizontal position in the zoomed window in seconds."""
+
+        if offset is not None:
+            self.timebase_offset = offset
+        if scale is not None:
+            self.timebase_scale = scale
+        if hor_magnify is not None:
+            self.timebase_hor_magnify = hor_magnify
+        if hor_position is not None:
+            self.timebase_hor_position = hor_position
+
+    ###############
+    # Acquisition #
+    ###############
+
+    acquisition_type = Instrument.control(
+        "ACQW?", "ACQW %s",
+        """ A string parameter that sets the type of data acquisition. Can be "normal", "peak",
+         "average", "highres".""",
+        validator=strict_discrete_set,
+        values={"normal": "SAMPLING", "peak": "PEAK_DETECT", "average": "AVERAGE", "highres": "HIGH_RES"},
+        map_values=True
+    )
+
+    acquisition_average = Instrument.control(
+        "AVGA?", "AVGA %d",
+        """ A integer parameter that selects the average times of average acquisition.""",
+        validator=strict_discrete_set,
+        values=[4, 16, 32, 64, 128, 256]
+    )
+
+    acquisition_status = Instrument.measurement(
+        "SAST?", """A string parameter that defines the acquisition status of the scope.""",
+        validator=strict_discrete_set,
+        values={"stopped": "Stop", "triggered": "Trig'd", "ready": "Ready", "auto": "Auto", "armed": "Arm"},
+        map_values=True
+    )
+
+    acquisition_sampling_rate = Instrument.measurement(
+        "SARA?", """A The SAST? sample rate of the scope."""
```

Comment:
Docstring mangled?

Reply:
Fixed in 795664353b97bf6e4db2098804f57a0e398c43a1